src/Authenticator/HttpDigestAuthenticator.php

```diff
@@ -100,7 +100,7 @@ public function authenticate(ServerRequestInterface $request, ResponseInterface
         }
 
         $hash = $this->generateResponseHash($digest, $password, $server['ORIGINAL_REQUEST_METHOD']);
-        if ($digest['response'] === $hash) {
+        if (hash_equals($digest['response'], $hash)) {
```

Comment:
arguments should be swapped

Reply:
```Note:
It is important to provide the user-supplied string as the second parameter, rather than the first.```

http://php.net/manual/en/function.hash-equals.php

Reply:
Thanks I missed that.